supercollider does not build on unstable

### Steps to reproduce

nix-shell -p supercollider

## Technical details

[anders@gurney:~]
$ nixos-version
16.09.git.ef7e753 (Flounder)
[anders@gurney:~]
$ nix-env --version
nix-env (Nix) 1.11.2
[anders@gurney:~]
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"16.09.git.ef7e753M"

## Error message

```
...
[ 34%] Linking CXX executable scsynth
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::runtime_error::runtime_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::size() const@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::append(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare(char const*) const@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&)@GLIBCXX_3.4.21'
/nix/store/jdix6k80a74a4548m9rzxaq1llm64vs6-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()@GLIBCXX_3.4.21'
collect2: error: ld returned 1 exit status
server/scsynth/CMakeFiles/scsynth.dir/build.make:101: recipe for target 'server/scsynth/scsynth' failed
make[2]: *** [server/scsynth/scsynth] Error 1
CMakeFiles/Makefile2:3331: recipe for target 'server/scsynth/CMakeFiles/scsynth.dir/all' failed
make[1]: *** [server/scsynth/CMakeFiles/scsynth.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/a2hyjaz0nh2vzdnxn5938gkrjw63qc6f-supercollider-3.6.6.drv’ failed with exit code 2
error: build of ‘/nix/store/a2hyjaz0nh2vzdnxn5938gkrjw63qc6f-supercollider-3.6.6.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

Same here:
### command
```
nix-env -i supercollider
```
### environment
```
~ nixos-version
16.09pre83796.d541e0d (Flounder)
~ nix-env --version
nix-env (Nix) 1.12pre4523_3b81b26
```


### error message
```
[ 31%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/SC_Reply.cpp.o
[ 31%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/SC_Sem.cpp.o
[ 32%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/SC_StandAloneInfo_Darwin.cpp.o
[ 32%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/SC_StringBuffer.cpp.o
[ 32%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/SC_StringParser.cpp.o
[ 32%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/scsynthsend.cpp.o
[ 32%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/__/__/common/sc_popen.cpp.o
[ 32%] Building C object server/scsynth/CMakeFiles/libscsynth.dir/__/__/external_libraries/TLSF-2.4.6/src/tlsf.c.o
[ 33%] Building CXX object server/scsynth/CMakeFiles/libscsynth.dir/SC_Jack.cpp.o
/tmp/nix-build-supercollider-3.6.6.drv-0/SuperCollider-Source/server/scsynth/SC_Jack.cpp: In member function 'bool SC_JackDriver::GraphOrderChanged()':
/tmp/nix-build-supercollider-3.6.6.drv-0/SuperCollider-Source/server/scsynth/SC_Jack.cpp:683:28: warning: 'jack_nframes_t jack_port_get_total_latency(jack_client_t*, jack_port_t*)' is deprecated (declared at /nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/include/jack/jack.h:1223) [-Wdeprecated-declarations]
   jack_nframes_t portLat = jack_port_get_total_latency(mClient, outputs->mPorts[i]);
                            ^
/tmp/nix-build-supercollider-3.6.6.drv-0/SuperCollider-Source/server/scsynth/SC_Jack.cpp:683:83: warning: 'jack_nframes_t jack_port_get_total_latency(jack_client_t*, jack_port_t*)' is deprecated (declared at /nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/include/jack/jack.h:1223) [-Wdeprecated-declarations]
   jack_nframes_t portLat = jack_port_get_total_latency(mClient, outputs->mPorts[i]);
                                                                                   ^
[ 33%] Linking CXX static library libscsynth.a
[ 33%] Built target libscsynth
Scanning dependencies of target scsynth
[ 34%] Building CXX object server/scsynth/CMakeFiles/scsynth.dir/scsynth_main.cpp.o
[ 34%] Linking CXX executable scsynth
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::c_str() const@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::runtime_error::runtime_error(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::size() const@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::append(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::compare(char const*) const@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(char const*, std::allocator<char> const&)@GLIBCXX_3.4.21'
/nix/store/l009qiysqg6gdvizhl37h0d1p2qrbzma-libjack2-1.9.10/lib/libjack.so: undefined reference to `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::~basic_string()@GLIBCXX_3.4.21'
collect2: error: ld returned 1 exit status
server/scsynth/CMakeFiles/scsynth.dir/build.make:101: recipe for target 'server/scsynth/scsynth' failed
make[2]: *** [server/scsynth/scsynth] Error 1
CMakeFiles/Makefile2:3331: recipe for target 'server/scsynth/CMakeFiles/scsynth.dir/all' failed
make[1]: *** [server/scsynth/CMakeFiles/scsynth.dir/all] Error 2
Makefile:160: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/r5j4p5sqd2bg0m9ha9qvnbkqa11jkbw5-supercollider-3.6.6.drv’ failed with exit code 2
error: build of ‘/nix/store/r5j4p5sqd2bg0m9ha9qvnbkqa11jkbw5-supercollider-3.6.6.drv’ failed

```

--- a/pkgs/development/interpreters/supercollider/default.nix
+++ b/pkgs/development/interpreters/supercollider/default.nix
@@ -3,10 +3,12 @@
 , libXt, qt, readline
 , useSCEL ? false, emacs
 }:
-  
-let optional = stdenv.lib.optional; in
 
-stdenv.mkDerivation rec {  
+let optional = stdenv.lib.optional;
+ljack2 = libjack2.override { gcc = gcc; };
+in
+
+stdenv.mkDerivation rec {
   name = "supercollider-3.6.6";
 
   meta = {
@@ -29,12 +31,12 @@ stdenv.mkDerivation rec {
 
   cmakeFlags = ''
     -DSC_WII=OFF
-    -DSC_EL=${if useSCEL then "ON" else "OFF"} 
+    -DSC_EL=${if useSCEL then "ON" else "OFF"}
   '';
 
   nativeBuildInputs = [ cmake pkgconfig ];
 
-  buildInputs = [ 
-    gcc libjack2 libsndfile fftw curl libXt qt readline ]
+  buildInputs = [
+    gcc ljack2 libsndfile fftw curl libXt qt readline ]
     ++ optional useSCEL emacs;
 }

--- a/pkgs/misc/jackaudio/default.nix
+++ b/pkgs/misc/jackaudio/default.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchFromGitHub, pkgconfig, python, makeWrapper
-, bash, libsamplerate, libsndfile, readline
+, bash, libsamplerate, libsndfile, readline, gcc
 
 # Optional Dependencies
 , dbus ? null, pythonDBus ? null, libffado ? null, alsaLib ? null
@@ -33,7 +33,7 @@ stdenv.mkDerivation rec {
   };
 
   nativeBuildInputs = [ pkgconfig python makeWrapper ];
-  buildInputs = [
+  buildInputs = [ gcc
     python
 
     libsamplerate libsndfile readline

